cypress/support/elements/iframe.js

```diff
@@ -1,21 +1,27 @@
-export const getIframeDocument = (outerSelector) => {
+export const getInIframe = (outerSelector, innerSelector) => {
+  let $innerResult;
+
   return cy.get(outerSelector)
-  .get("iframe")
-  // Cypress yields jQuery element, which has the real
-  // DOM element under property "0".
-  // From the real DOM iframe element we can get
-  // the "document" element, it is stored in "contentDocument" property
-  // Cypress "its" command can access deep properties using dot notation
-  // https://on.cypress.io/its
-  .its("0.contentDocument").should("exist");
-};
+  // First make sure the innerSelector exists by wrapping the whole search
+  // for it in a should function. If anything in the should function throws an
+  // exception or fails a test then the should function will automatically re-run
+  .should(($el) => {
+    // direct jQuery is used here because the cypress command for wrap, find, and its
+    // run asynchronously, and create confusing log message.
+
+    // Get the body element of the first iframe
+    const $firstIframe = $el.find("iframe")
+    // This line might throw an exception because the iframe doesn't exist or isn't
+    // loaded yet.  But in that case should will automatically retry
+    const iframeBody = $firstIframe[0].contentDocument.body;
+    $innerResult = Cypress.$(iframeBody).find(innerSelector);
 
-export const getIframeBody = (outerSelector) => {
-  // get the document
-  return getIframeDocument(outerSelector)
-  // automatically retries until body is loaded
-  .its("body").should("not.be.undefined")
-  // wraps "body" DOM element to allow
-  // chaining more Cypress commands, like ".find(...)"
-  .then(cy.wrap);
-};
+    // Make sure the innerSelect actually finds something
+    // this will print a message to the cypress log
+    expect($innerResult).to.exist;
```

Comment:
Good point I just double checked. It is this matcher:
https://github.com/chaijs/chai-jquery#exist

Which says it: "Assert that the selection is not empty".

It does seems to work correctly because otherwise this code would not have helped the test pass without the long wait that was required before.